PlanningPoker/ClientApp/src/app/room/room.component.ts

```diff
@@ -21,74 +21,79 @@ export class RoomComponent implements OnInit {
   public users: string[] = [];
   id: string;
   roles: string[] = [];
+  @HostListener('window:beforeunload') goToPage() {
+    this.userService.deleteUserFromRoom();
+    sessionStorage.setItem('Unload', '1');
+  }
   ngOnInit(): void {
+    if (sessionStorage.getItem('Unload')) {
+      this.router.navigate(['']);
+      sessionStorage.removeItem('Unload');
+    }
     this.votesCount = 0;
-    this.userService.getRoles(this.id);
-    this.userService.roles.subscribe((role) => {
-      this.role = role;
-    });
-    this.userService.getUsers(this.id).subscribe(users => {
-      this.users = [];
-      // tslint:disable-next-line:forin
-      for (const user in users) {
-        this.users.push(users[user]);
-      }
-      this.userService.getRolesForRoom(this.users, this.id).subscribe((roles) => {
-        for (const role in roles) {
-          if (role) {
-            this.roles.push(roles[role]);
-          }
-        }
-      });
-    });
-    this.userService.finishVoting
-      .subscribe(() => {
-        this.userService.getUserVote(this.id).subscribe(result => {
-          this.userVote = [];
-          for (const user in result) {
-            if (user) {
-              const index = this.users.indexOf(user);
-              this.userVote[index] = result[user];
-            }
-          }
-        }, error => console.error(error));
-        console.log(this.userVote);
-      });
-    this.userService.cleared
-      .subscribe(
-        () => {
-          this.Votes = [];
-          this.userVote = [];
-          this.votesCount = 0;
-          localStorage.removeItem('UserVote');
-          this.users = [];
-          this.userService.getUsers(this.id).subscribe(users => {
-            this.users = [];
-            // tslint:disable-next-line:forin
-            for (const user in users) {
-              this.users.push(users[user]);
-            }
-          });
-        });
     this.userService.join
       .subscribe(
         () => {
           this.userService.getUsers(this.id).subscribe(users => {
             this.users = [];
             // tslint:disable-next-line:forin
             for (const user in users) {
-              this.users.push(users[user]);
+              this.users.push(users[user]['name']);
             }
             this.userService.getRolesForRoom(this.users, this.id).subscribe((roles) => {
               this.roles = [];
+              // tslint:disable-next-line:forin
               for (const role in roles) {
```

Comment:
you can use for of loop, instead of for in, it will allow you to remove tslint hint